src/main/resources/org/broadinstitute/hellbender/tools/walkers/vqsr/nvscorevariants.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python3
+
+import argparse
+import torch
+import sys
+import pytorch_lightning as pl
+from lightning_fabric.utilities import seed
+from torch.utils.data import DataLoader
+from scorevariants.dataset import ReferenceDataset
+from scorevariants.readers import TensorReader, ReferenceTensorReader
+from scorevariants.models.wrapper import LightningWrapper
+from scorevariants.create_output_vcf import create_output_vcf
+
+def get_model(args, model_file):
+    """
+    Args:
+        args: model specific and trainer arguments
+        model_file: model file to load parameters from a pretrained PyTorch model
+    """
+    dict_args = vars(args)
+    model = torch.load(model_file)
+    model = LightningWrapper(model, **dict_args)
+    return model
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--vcf-file', required=True, help='A VCF file containing variants to creat datasets')
+    parser.add_argument('--ref-file', required=True, help='Reference sequence file to creat datasets')
+    parser.add_argument('--input-file', help='BAM file containing reads to creat datasets')
+    parser.add_argument('--tensor-type', default='reference', help='Name of the tensors to generate, reference for 1D reference tensors and read_tensor for 2D tensors.')
+    parser.add_argument('--batch-size', type=int, default=32, help='Batch size')
+    parser.add_argument('--seed', type=int, default=724, help='Seed to initialize the random number generator')
+    parser.add_argument('--tmp-file', default='tmp.txt', help='The temporary VCF-like file where variants scores will be written')
+    parser.add_argument('--output-file', required=True, help='Output VCF file')
+    parser.add_argument('--gpus', type=int, nargs='+', help='Number of GPUs (int) or which GPUs (list)')
+    parser.add_argument('--model-directory', default='models', help='Directory containing model files')
+    args = parser.parse_args()
+    return args
+
+def main():
+    args = get_args()
+    torch.manual_seed(args.seed)
+    seed.seed_everything(args.seed)
+
+    if args.tensor_type == 'reference':
+        model_file = args.model_directory + '/1d_cnn_mix_train_full_bn.pt'
+        tensor_reader = ReferenceTensorReader.from_files(args.vcf_file, args.ref_file)
+        label = 'CNN_1D'
+    elif args.tensor_type == 'read_tensor':
+        model_file = args.model_directory + '/small_2d.pt'
+        tensor_reader = TensorReader.from_files(args.vcf_file, args.ref_file, args.input_file)
+        label = 'CNN_2D'
+    else:
+        sys.exit('Unknown tensor type!')
+    model = get_model(args, model_file)
+    trainer = pl.Trainer(gradient_clip_val=1.0)
```

Comment:
I think I figured out the `add_argparse_args` thing.  If I understand [the docs](https://lightning.ai/docs/pytorch/1.8.6/common/trainer.html#trainer-in-python-scripts) correctly, it's basically there to enable you to skip writing out an exhaustive `ArgumentParser`.  Basically, it makes it so if you, as a person running this tool, use some args that aren't explicitly added to the `ArgumentParser`, but they are in the list of args that can be passed to a `Trainer`, lightning says "Oh, I recognize these," and then handles them as if you'd manually defined them.  I think now, again if I'm reading [the docs](https://lightning.ai/docs/pytorch/2.4.0/common/trainer.html#trainer-in-python-scripts) correctly, that's the default behavior for the `ArgumentParser`, but we might have to explicitly pass them to the trainer?  That doc section I linked doesn't include an example that makes that super clear.

Reply:
This comment implies that perhaps we should pass `**vars(args)` in to the Trainer here, but only if we actually want to expose any actual Trainer arguments as CLI arguments: https://lightning.ai/forums/t/how-to-combine-ptl-arguments-with-argumentparser/2440/2

Reply:
Oh, that's kind of neat.  Not how GATK arg parsing works.  So it would be hard to match that behavior in the GATK front end.  I guess we could have an "--additional args" that just get passed through?

Reply:
In the new framework (as that comment explains), you need to explicitly add any arguments you want to expose to the Python arg parser.